extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java

```diff
@@ -102,16 +104,26 @@ public JpaEntitiesBuildItem discoverModelAndRegisterForReflection() throws IOExc
             }
         }
 
+        if (!unindexedClasses.isEmpty()) {
+            final String unindexedClassesErrorMessage = unindexedClasses.stream().map(d -> "\t- " + d + "\n")
+                    .collect(Collectors.joining());
+            throw new ConfigurationError(
```

Comment:
I don't know if this should cause the augmentation to fail. This only really affects native mode, so you might prevent an otherwise valid app from running in JVM mode, when the end user has no intention of actually using native mode.

Reply:
@stuartwdouglas good remark. We should probably log a warning message instead of throwing?

Reply:
I know @dmlloyd was thinking about making JVM and native as close as possible so that when your app is working in JVM mode, you can be fairly confident it works in native mode.

BTW, it does not only register the classes for reflection, it also collects the list of domain objects.

Reply:
I saw the [discussion](https://github.com/quarkusio/quarkus/issues/1605#issuecomment-475217200) @gsmet was referring to. As a dev, I'd like the kind of consistency and assurance that "it works everywhere" brings in my development process.